libtenzir/src/tql2/eval_unary.cpp

```diff
@@ -88,6 +88,34 @@ struct EvalUnOp<ast::unary_op::neg, T> {
   }
 };
 
+template <>
+struct EvalUnOp<ast::unary_op::neg, duration_type> {
+  using T = type_to_arrow_array_t<duration_type>;
+  static auto eval(const T& x, auto warn) -> std::shared_ptr<T> {
+    auto b = duration_type::make_arrow_builder(arrow::default_memory_pool());
+    check(b->Reserve(x.length()));
+    auto overflow = false;
+    for (auto i = int64_t{0}; i < x.length(); ++i) {
+      if (x.IsNull(i)) {
+        check(b->AppendNull());
+        continue;
+      }
+      auto val = x.Value(i);
+      static_assert(sizeof(val) == sizeof(duration::rep));
+      if (val == std::numeric_limits<duration::rep>::min()) {
+          overflow = true;
+          check(b->AppendNull());
+          continue;
+      }
+      check(b->Append(-val));
+      if (overflow) {
+        warn("Duration overflowed");
+      }
+    }
```

Comment:
```suggestion
    }
    if (overflow) {
      warn("duration negation overflow");
    }
```

The point of the flag is to emit the warning once, outside of the loop. Also, not sure whether you omitted the operation on purposed (you have it in the other strings).

Reply:
Nope, just missed this one. Thanks for spotting!